### ISSUE TITLE
another command to use `ffip-diff-*`

### ISSUE THREAD

**Issue by redguardtoo** [technomancy/find-file-in-project#78]
create a buffer containing diff content, set default-directory (check abo-abo ivy code),
enable `ffip-diff-mode`

**Comment by sten0** [comment]
On a closely related topic, I've been having trouble getting ffip-ffip-show-diff to pass on Debian's Continuous Integration infrastructure.  Self-tests pass fine on the buildd infrastructure after a1ec308c6427014266b4a385c63c0c9c83cbc70e, but CI seems to be even more strict, and more limited, so I put in the time to try to run them without test.sh.

So far I've failed.  I think the problem is ido-completing-read sets default-directory back to ./ (eg: find-file-in-project).  The strange thing is set default-directory didn't work with dh_elpa's ert testing, but (cd "tests") did.  I uploaded this branch here:
https://github.com/sten0/find-file-in-project/tree/experiment

BUILD-LOG-BACKTRACE has the backtrace.  I wonder if this (#78) issue will also run into trouble with default-directory (whether on your end, or on Continous Integration).  At any any rate, I hope my experiment failed in a way that's useful ;-)

**Comment by redguardtoo** [comment]
It's fine to use `cd` instead of setup `default-directory` in test cases. If you only change test cases, I'm OK to review and accept your pull request asap.

**Comment by sten0** [comment]
When reading through the commit history I found 8eedc930ec7fb187ebc3cec284e213054545db09, and after reading that I realised why the CI tests were failing, and also that a1ec308c6427014266b4a385c63c0c9c83cbc70e is no longer necessary.

Find-file-in-project now has continuous integration testing :-)  Here's the link for current status and logs:
https://ci.debian.net/packages/f/find-file-in-project/

In conclusion, there seems to be something in Debian CI that prevents setting default-directory or ffip-project-root, but methods that depend on ```cd``` are ok, and getting default-directory works.

I believe my modifications to how tests are run are almost certainly Debian-specific, but here is the source for our package:
https://anonscm.debian.org/git/pkg-emacsen/pkg/find-file-in-project.git

**Comment by redguardtoo** [comment]
Good to know. Thanks.

**Comment by sten0** [comment]
You're welcome.

P.S. Re: abo-abo ivy code, Ivy/swiper tests run 100% fine out of the box, with no special configuration, so I'm guessing they're a good model :-)
https://ci.debian.net/packages/e/emacs-ivy/
https://buildd.debian.org/status/package.php?p=emacs-ivy
